### PR TITLE
[WIP] y2packager license API

### PR DESCRIPTION
**Pull request by teclator**
- [Trello Card](https://trello.com/c/Ve9xuqt3/1377-8-licenses-help-customers-to-understand-that-they-are-accepting-another-license-for-another-product-no-you-know-what-just-make-i)

### CODE REVIEW

File: library/packages/src/lib/y2packager/licenses_fetchers/base.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # Base class for licences fetchers
+    class Base
+      include Yast::Logger
+
+      # @return [String] Product name to get licenses for
+      attr_reader :product_name
+
+      # Constructor
+      #
+      # @param product_name [String] to get licenses for
```
Comment (teclator) [inline library/packages/src/lib/y2packager/licenses_fetchers/base.rb]: It should be a `Hash`

File: library/packages/src/lib/y2packager/license.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  # for different languages.
+  class License
+    DEFAULT_LANG = "en_US".freeze
+    # @return [String] License unique identifier
+    attr_reader :id
+
+    # @return [Boolean] whether the license has been accepted or not
+    attr_reader :accepted
+
+    # @retrun [Hash<String, String>] language -> content
```
Comment (imobachgs) [inline library/packages/src/lib/y2packager/license.rb]: return

File: library/packages/src/lib/y2packager/license.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    # Set the license as rejected
+    def reject!
+      @accepted = false
+    end
+
+  private
+
+    # Generate the license unique identifier based on the given options.
+    # Currently the id is obtained using the MD5 digest of te license's
```
Comment (imobachgs) [inline library/packages/src/lib/y2packager/license.rb]: te -> the

File: library/packages/src/lib/y2packager/license_reader.rb [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "y2packager/license"
+require "y2packager/license_store"
+require "y2packager/licenses_fetchers/rpm"
+require "y2packager/licenses_fetchers/url"
+
+module Y2Packager
+  # This class is able to read the licence of a given product
```
Comment (imobachgs) [inline library/packages/src/lib/y2packager/license_reader.rb]: licence -> license

File: library/packages/src/lib/y2packager/license_reader.rb [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    include Yast::Logger
+
+    attr_reader :product_name
+    attr_reader :source
+
+    # Constructor
+    #
+    # @param product_name [String] Product name to get licenses for
+    # @param source       [Symbol] Source to use for fetching the license from
+    def initialize(product_name, source = :rpm)
```
Comment (imobachgs) [inline library/packages/src/lib/y2packager/license_reader.rb]: I would rename `:rpm` to `:libzypp`, but it is just a matter of taste.

**Comment by teclator** [comment]
This PR has been replaced by #718

**Pull request closed without merging**